zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -234,3 +234,60 @@ impl GetBlockTemplate {
         }
     }
 }
+
+/// Error response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ProposalRejectReason {
+    /// Block rejected as invalid
+    Rejected,
+}
+
+/// Response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged, rename_all = "kebab-case")]
+pub enum ProposalResponse {
+    /// Block was not successfully submitted, return error
+    ErrorResponse {
+        /// Reason the proposal was invalid as-is
+        reject_reason: ProposalRejectReason,
+
+        /// The getblocktemplate RPC capabilities supported by Zebra.
+        capabilities: Vec<String>,
+    },
```

Comment:
Please link to the documentation that lists the fields in this JSON-RPC response.

Reply:
Added in [`8420b75`](https://github.com/ZcashFoundation/zebra/pull/5870/commits/8420b75013c1a05ea11a9799d1f87bd338c3c27c).